src/internal/api/authenticator.go

```diff
@@ -76,11 +74,21 @@ func (a *authenticator) Login(userName, password string) error {
 		return err
 	}
 
-	if err = a.cfgSvc.SetCredentials(creds.AccessToken, creds.RefreshToken, creds.ExpiresIn); err != nil {
-		return fmt.Errorf("failed to save credentials in storage: %w", err)
+	accessTokenExpDate, err := jwt.ExpirationDate(creds.AccessToken)
+	if err != nil {
+		return fmt.Errorf("failed to extract expiration date from access token: %w", err)
 	}
 
-	a.status = statusWorkingProperly
+	refreshTokenExpDate, err := jwt.ExpirationDate(creds.RefreshToken)
+	if err != nil {
+		return fmt.Errorf("failed to extract expiration date from access token: %w", err)
+	}
+
+	if err = a.cfgSvc.SetCredentials(
+		creds.AccessToken, creds.RefreshToken, accessTokenExpDate, refreshTokenExpDate,
+	); err != nil {
+		return fmt.Errorf("failed to save credentials in storage: %w", err)
+	}
```

Comment:
Please add `a.backoff.Reset()` call at the end of the process just to be sure we don't have any leftovers.